platform/security/idp/security-idp-server/src/main/java/org/codice/ddf/security/idp/server/IdpEndpoint.java

```diff
@@ -1331,12 +1334,16 @@ private Response getSamlRedirectResponse(
             RestSecurity.deflateAndBase64Encode(
                 DOM2Writer.nodeToString(OpenSAMLUtil.toDom(samlResponse, doc, false))),
             "UTF-8");
-    String requestToSign =
-        String.format("%s=%s&RelayState=%s", samlType.getKey(), encodedResponse, relayState);
+    StringBuilder requestToSign =
+        new StringBuilder(samlType.getKey()).append("=").append(encodedResponse);
+    if (StringUtils.isNotBlank(relayState)) {
+      requestToSign.append("&RelayState=").append(relayState);
+    }
     UriBuilder uriBuilder = UriBuilder.fromUri(targetUrl);
     uriBuilder.queryParam(samlType.getKey(), encodedResponse);
-    uriBuilder.queryParam(SSOConstants.RELAY_STATE, relayState == null ? "" : relayState);
-    new SimpleSign(systemCrypto).signUriString(requestToSign, uriBuilder);
+    uriBuilder.queryParam(
+        SSOConstants.RELAY_STATE, StringUtils.isBlank(relayState) ? "" : relayState);
+    new SimpleSign(systemCrypto).signUriString(requestToSign.toString(), uriBuilder);
```

Comment:
❓ Should we add or modify any unit tests in `IdpEndpointSpecTest` to account for these changes?